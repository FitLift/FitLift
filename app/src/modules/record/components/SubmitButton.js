import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
  TouchableHighlight,
  Text,
  View,
} from 'react-native';

export default class NumberInput extends PureComponent {
  static propTypes = {
    submitButtonColor: PropTypes.string.isRequired,
  };

  onPress = (e) => {
    console.log(e);
  }

  render() {
    const {
      submitButtonColor,
    } = this.props;
    return (
      <TouchableHighlight
        underlayColor="white"
        onPress={this.onPress}
      >
        <View style={{
          backgroundColor: submitButtonColor,
          borderRadius: 4,
          height: 30,
          justifyContent: 'center',
          padding: 5,
        }}
        >
          <Text style={{
            color: 'white',
          }}
          >
            Submit
          </Text>
        </View>
      </TouchableHighlight>
    );
  }
}
