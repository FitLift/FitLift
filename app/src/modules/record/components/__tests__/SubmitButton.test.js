import React from 'react';
import renderer from 'react-test-renderer';
import SubmitButton from '../SubmitButton';

describe('render tests', () => {
  it('doesnt crash', () => {
    const rendered = renderer.create(
      <SubmitButton
        onPress={f => f}
        submitButtonColor="#9CCC65"
      />,
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
