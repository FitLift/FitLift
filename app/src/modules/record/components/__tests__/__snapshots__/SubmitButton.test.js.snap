// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tests doesnt crash 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "#9CCC65",
        "borderRadius": 4,
        "height": 30,
        "justifyContent": "center",
        "padding": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
